arrow-flight/src/decode.rs

```diff
@@ -101,13 +97,12 @@ impl FlightRecordBatchStream {
     {
         Self {
             inner: FlightDataDecoder::new(inner),
-            got_schema: false,
         }
     }
 
-    /// Has a message defining the schema been received yet?
-    pub fn got_schema(&self) -> bool {
-        self.got_schema
+    /// Return schema for the stream, if it has been received
+    pub fn schema(&self) -> Option<SchemaRef> {
```

Comment:
```suggestion
    /// Return schema for the stream, if it has been received
    pub fn schema(&self) -> Option<&SchemaRef> {
```

Returning owned types is a code smell in most cases, as it breaks borrow propagation